Statless authz update (#221)
Altinn/altinn-studio-docs: content/teknologi/altinnstudio/changelog/app-nuget/v4/whats-new.md

toc: true
---

### 4.7.0 (2021-06-08)

Altinn Apps now authorize access for statless apps.

> Altinn Apps have now two new Application Events where application developers can add data processing logic. Calulation, population +++

Reviewer: Instead of +++ we could write "and more", or something similar.
```suggestion
Altinn Apps now have two new application events where application developers can add data processing logic. calulation, population, and more.
```

> In addition, RunCalculate Application event is now OBSOLETE. It is recommended  that Apps should move from RunCalculation to RunProcessDataWrite and RunProcessDataRead.
> In the future, RunCalculate methods will not be supported.

Reviewer: ```suggestion
In this update the RunCalculate application event is made obsolete/deprecated. It's recommended that Apps are updated to use RunProcessDataWrite and RunProcessDataRead instead. Calls to the RunCalculate method will be removed in a future update.
```

The process to update is

> 1. Add DataProcessing folder and Handler from template [here](https://github.com/Altinn/altinn-studio/tree/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic)

Reviewer: ```suggestion
1. Add the DataProcessing folder and DataProcessingHandler class from our [app template](https://github.com/Altinn/altinn-studio/tree/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic) to your app.
```

> 2. Update App.cs. Add referane to dataprocessinghandler and copy new methods from [App.cs](https://github.com/Altinn/altinn-studio/blob/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic/App.cs#L112)

Reviewer 2: Will these line number references be future proof? Won't we risk that things might be added and line number don't apply?

Author: fixed

> 2. Update App.cs. Add referane to dataprocessinghandler and copy new methods from [App.cs](https://github.com/Altinn/altinn-studio/blob/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic/App.cs#L112)

Reviewer: ```suggestion
2. Update App.cs. Add a class field for DataProcessingHandler and copy new methods from [App.cs](https://github.com/Altinn/altinn-studio/blob/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic/App.cs#L112)
```

3. Move logic from calculation handler to DataProcessinghandler
4. Remove RunCalculation method from App.cs

> 5. Remove calculationHandler when code is moved to DataProcessingHandler

Reviewer: ```suggestion
5. Remove CalculationHandler when code has been moved to DataProcessingHandler.
```

6. Compile and test your app. 

> See details about datataprocessing [here](https://altinn.github.io/docs/altinn-studio/app-creation/logic/dataprocessing/)

Reviewer: ```suggestion
See details about data processing [here](https://altinn.github.io/docs/altinn-studio/app-creation/logic/dataprocessing/)
```

## 4.7.1 (2021-06-15) - Adjustments to response headers
Some of the controllers exposed by the applications have been modified to not allow caching and/or storage of their responces in the client.

### 4.7.0 (2021-06-08)

Altinn Apps now authorize access for statless apps.

Altinn Apps have now two new Application Events where application developers can add data processing logic. Calulation, population +++

In addition, RunCalculate Application event is now OBSOLETE. It is recommended  that Apps should move from RunCalculation to RunProcessDataWrite and RunProcessDataRead.
In the future, RunCalculate methods will not be supported.

The process to update is

1. Add DataProcessing folder and Handler from template [here](https://github.com/Altinn/altinn-studio/tree/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic)

> 2. Update App.cs. Add referane to dataprocessinghandler and copy new methods ( RunProcessDataRad and RunProcessDataWrite) from [App.cs](https://github.com/Altinn/altinn-studio/blob/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic/App.cs)

Reviewer: ```suggestion
2. Update App.cs. Add new field for DataProcessingHandler and copy new methods ( RunProcessDataRad and RunProcessDataWrite) from [App.cs](https://github.com/Altinn/altinn-studio/blob/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic/App.cs)
```
